CodeReviewGabe.cpp

```diff
@@ -0,0 +1,276 @@
+
+      splitString(placeHolder[9], delimiter2, friends, index2);  
+      user2.friends[0] = friends[2];
+
+      //User 3 Friends
+      placeHolder[14].erase(remove(placeHolder[14].begin(), placeHolder[14].end(), '['), placeHolder[14].end()); //remove [ from string
+      placeHolder[14].erase(remove(placeHolder[14].begin(), placeHolder[14].end(), ']'), placeHolder[14].end()); //remove ] from string
+
+      splitString(placeHolder[14], delimiter2, friends, index2);  
+      user3.friends[0] = friends[3];
+      inFile.close();
+      // for(int i = 0; i < 100; i++){
+      //   cout << placeHolder[i] << endl;
+      // }
+
+      
+      //Load Post Data
+      cout << "Enter Post.txt Filename: ";
+      string filename1;
+      cin >> filename1;
+
+      //Handle Post File
+      inFile.open(filename1);
+
+      // Test for errors.
+      if (!inFile) {
+          cout << "Error opening the file.\n";
+          return 0;
+      }
+
+      //Assign Posts
+      index2 = 0;
+      inFile >> line;
+      splitString(line, delimiter, placeHolder2, index2);
+      post1.postID = placeHolder2[0];
+      post1.userID = placeHolder2[1];
+      post1.visibility = placeHolder2[2];
+
+      index2 = 0;
+      inFile >> line;
+      splitString(line, delimiter, placeHolder2, index2);
+      post2.postID = placeHolder2[0];
+      post2.userID = placeHolder2[1];
+      post2.visibility = placeHolder2[2];
+
+      index2 = 0;
+      inFile >> line;
+      splitString(line, delimiter, placeHolder2, index2);
+      post3.postID = placeHolder2[0];
+      post3.userID = placeHolder2[1];
+      post3.visibility = placeHolder2[2];
+    }
+      
+    else if(userChoice == 2){
+      string postID;
+      string username;
+      string visibility;
+      
+      cout << "Enter a Post ID: " << endl;
+      cin >> postID;
+
+      cout << "Enter a UserName: " << endl;
+      cin >> username;
+
+      if(postID == post1.postID){
+        visibility = post1.visibility;
+        //If visibility is friend, check user's friends and make sure they match given username
+        if(visibility == "friend"){
+          for(int i = 0; i < 2; i++){
+            if(user1.friends[i] == username){
+              cout << "Access Permitted" << endl;
+            }
+            else{
+              cout << "Access Denied" << endl;
+            }
+          }
+        }
+      }
+      else if(postID == post2.postID){
+        visibility = post2.visibility;
+        //If visibility is friend, check user's friends and make sure they match given username
+        if(visibility == "friend"){
+          for(int i = 0; i < 1; i++){
+            if(user2.friends[i] == username){
+              cout << "Access Permitted" << endl;
+            }
+            else{
+              cout << "Access Denied" << endl;
+            }
+          }
+        }
+      }
+      else if(postID == post3.postID){
+        cout << "Access Permitted" << endl;
```

Comment:
This seems very specific to the test case we were given, it might be better to check if `post3.visibility == "public"`